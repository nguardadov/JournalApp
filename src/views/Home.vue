<template>
  <div class="home">
    <img alt="Vue logo" src="../assets/logo.png" />
  </div>

  <button class="btn btn-primary" @click="goDayBook">primary</button>
  <button class="btn btn-secondary" @click="goDayBook">secondary</button>
  <button class="btn btn-success" @click="goDayBook">success</button>
  <button class="btn btn-warning" @click="goDayBook">warning</button>
</template>

<script>
export default {
  methods: {
    goDayBook() {
      this.$router.push({ name: 'no-entry' })
    },
  },
}
</script>
