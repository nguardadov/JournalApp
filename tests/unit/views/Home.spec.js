import { shallowMount } from '@vue/test-utils'
import Home from '@/views/Home.vue'

describe('Pruebas en el Home view', () => {
  test('debe de renderizar el componente correctament', () => {
    const wrapper = shallowMount(Home)
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('Hacer click en un boton debe de redireccionar a no-entry', () => {
    const mockRouter = {
      push: jest.fn(),
    }

    //aca lo que hacemos es indicarle que use nuestro mock y no el global que trae el $router
    const wrapper = shallowMount(Home, {
      global: {
        mocks: {
          $router: mockRouter,
        },
      },
    })

    wrapper.find('button').trigger('click')
    expect(mockRouter.push).toHaveBeenCalled()
    expect(mockRouter.push).toBeCalledWith({ name: 'no-entry' })
  })
})
