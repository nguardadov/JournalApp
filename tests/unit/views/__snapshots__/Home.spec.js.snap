// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el Home view debe de renderizar el componente correctament 1`] = `<div class="home"><img alt="Vue logo" src=""></div><button class="btn btn-primary">primary</button><button class="btn btn-secondary">secondary</button><button class="btn btn-success">success</button><button class="btn btn-warning">warning</button>`;
